wgengine/magicsock/magicsock.go

```diff
@@ -918,6 +918,13 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 			addAddr(netip.AddrPortFrom(v4Addrs[0].Addr(), uint16(port)), tailcfg.EndpointSTUN4LocalPort)
 		}
 	}
+	if s := envknob.String("TS_DEBUG_PRETENDPOINT"); s != "" {
+		// Temporarily (2024-07-08) during investigations, allow setting
+		// a pretend endpoint for testing NAT traversal scenarios.
+		// TODO(bradfitz): probably promote this to the config file.
+		// https://github.com/tailscale/tailscale/issues/12578
+		addAddr(ipp(s), tailcfg.EndpointExplicitConf)
```

Comment:
`ipp` hides parse errors; given that these are user-provided, maybe worth logging an error for easier debugging?

Reply:
Done.

Reply:
Also moved it to debugknobs.go next to the other envknobs, and made it sync.OnceValue ... no need to parse it every time.